File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let hashtable = {};
    let neededvalue = 0;
    for(let i = 0;i<nums.length;i++){
        neededvalue = target - nums[i];
        if (neededvalue in hashtable)
        return [i,hashtable[neededvalue]]
        else
        hashtable[nums[i]] = i;
    }
    return [];
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(0, len(nums) - 1):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i,j]

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}
        for i in range(len(nums)):
            required = target - nums[i]
            if required in hash:
                return [i,hash[required]]
            else:
                hash[nums[i]] = i